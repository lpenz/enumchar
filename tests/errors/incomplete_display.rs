// Copyright (C) 2025 Leandro Lisboa Penz <lpenz@lpenz.org>
// This file is subject to the terms and conditions defined in
// file 'LICENSE', which is part of this source code package.

use enumchar::EnumChar;

#[derive(EnumChar)]
pub enum E {
    #[char('.')]
    Dot,
    Other,
}

fn main() {
    println!("{}", E::Dot);
}
