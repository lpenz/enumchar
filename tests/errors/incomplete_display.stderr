error[E0277]: `E` doesn't implement `std::fmt::Display`
  --> tests/errors/incomplete_display.rs:15:20
   |
15 |     println!("{}", E::Dot);
   |                    ^^^^^^ `E` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `E`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = note: this error originates in the macro `$crate::format_args_nl` which comes from the expansion of the macro `println` (in Nightly builds, run with -Z macro-backtrace for more info)
